macros/core/PGML.pl

```diff
@@ -1523,17 +1501,15 @@ sub Escape {
 }
 
 sub Indent {
-	my $self = shift;
-	my $item = shift;
+	my ($self, $item, $block, $state) = @_;
 	return $self->string($item) if $item->{indent} == 0;
-	my $em = 2.25 * $item->{indent};
-	return
-		$self->nl
-		. '<div style="margin:0 0 0 '
-		. $em . 'em">' . "\n"
-		. $self->string($item)
-		. $self->nl
-		. "</div>\n";
+	my $em     = 2.25 * $item->{indent};
+	my $bottom = '0';
+	if (defined $block->{stack}[ $state->{i} ] && $block->{stack}[ $state->{i} ]{type} eq 'par') {
```

Comment:
It might be useful to do

``` perl
my $next = $block->{stack}[ $state->{i} ];
if (defined $next && $next->type eq 'par') {
```

to reduce the number of dereferences that need to be performed.  Similarly for lines 1544 and 1558.